derp/limiter_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package derp
+
+import (
+	"testing"
+	"time"
+)
+
+func mb(mb uint64) uint64 {
+	return mb * 1024 * 1024
+}
+
+func TestLimiterLoopGradual(t *testing.T) {
```

Comment:
Make this a table-based test, with successive amounts of traffic and the expected limiter output, and verifies that at each step the loop settles on the expected output.

For bonus points, also have tests that shuffle the table and verify that everything still works, since the loop should settle on the same new output regardless of past values.